services/payments/statemachine_solana.go

```diff
@@ -198,16 +198,33 @@ func (sm *SolanaMachine) Pay(ctx context.Context) (*paymentLib.AuthenticatedPaym
 	}
 
 	b58Signature := base58.Encode(idempotencyData.Transaction.Signatures[0])
+	// solanaError stores temporary errors that need to be recorded as the LastError in QLDB but
+	// that are expected and do not need to be acted upon. It will only be returned if no other
+	// unexpected errors arise from processing.
+	var solanaError error
 	status, err := checkStatus(ctx, b58Signature, &sm.solanaRpcClient)
 	if err != nil {
-		return sm.transaction, fmt.Errorf("failed to check transaction status: %w", err)
+		// Some errors are expected and we just want to record them. Check if the status check
+		// returned such an error and prepare to return it after we attempt to make progress.
+		var (
+			errNotConfirmed SolanaTransactionNotConfirmedError
+			errNotFound     SolanaTransactionNotFoundError
+			errUnknown      SolanaTransactionUnknownError
+		)
+		if errors.As(err, &errNotConfirmed) ||
+			errors.As(err, errNotFound) ||
+			errors.As(err, errUnknown) {
```

Comment:
And this should be passing pointers too:

```suggestion
		if errors.As(err, &errNotConfirmed) || errors.As(err, &errNotFound) || errors.As(err, &errUnknown) {
```

But, as you can see, the approach is already problematic – you're having to create a bunch of variables instead of a simple check against a sentinel const-error (see my comment above). Here's how we handle that in SKUs – https://github.com/brave-intl/bat-go/blob/master/services/skus/controllers.go#L1167-L1207.